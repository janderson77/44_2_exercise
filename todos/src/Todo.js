import React from 'react'

const Todo = ({removeTodo, todoText, id}) => {
    const handleRemoveTodo = () => {
        removeTodo(id)
    }
    return(
        <div id={id} className="Todo">
            <p data-testid="Todo-text" className="Todo-text">{todoText}
            <button 
                id="Todo-removeBtn" 
                className="btn btn-danger"
                onClick={handleRemoveTodo}
            >Done!</button></p>
        </div>
    )
}

export default Todo