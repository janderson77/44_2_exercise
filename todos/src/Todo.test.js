import React from 'react';
import { render } from '@testing-library/react';
import {within} from '@testing-library/dom'
import Todo from './Todo'

let todo;
beforeEach(() => {
    todo = render(
        <Todo 
            todoText="This is a todo"
        />
    )
})

it('renders without crashing', () => {
    expect(todo).not.toBeNull()
})

it('matches snapshot', () => {
    expect(todo.asFragment()).toMatchSnapshot()
})

it('uses todoText prop to add appropriate text', () => {
    const todoText = todo.getByTestId('Todo-text')

    const {getByText} = within(todoText)
    const text = getByText('This is a todo')

    expect(text).toBeInTheDocument()
    expect
})