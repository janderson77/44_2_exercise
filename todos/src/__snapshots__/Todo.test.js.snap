// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <p
      class="Todo-text"
      data-testid="Todo-text"
    >
      This is a todo
      <button
        class="btn btn-danger"
        id="Todo-removeBtn"
      >
        Done!
      </button>
    </p>
  </div>
</DocumentFragment>
`;
